dlt/destinations/impl/synapse/synapse.py

```diff
@@ -0,0 +1,99 @@
+from typing import ClassVar, Sequence, List, Dict, Any, Optional
+from copy import deepcopy
+
+from dlt.common.destination import DestinationCapabilitiesContext
+from dlt.common.destination.reference import SupportsStagingDestination, NewLoadJob
+
+from dlt.common.schema import TTableSchema, TColumnSchema, Schema, TColumnHint
+from dlt.common.schema.typing import TTableSchemaColumns
+
+from dlt.destinations.sql_jobs import SqlStagingCopyJob, SqlJobParams
+from dlt.destinations.sql_client import SqlClientBase
+from dlt.destinations.insert_job_client import InsertValuesJobClient
+from dlt.destinations.job_client_impl import SqlJobClientBase
+
+from dlt.destinations.impl.mssql.mssql import MsSqlTypeMapper, MsSqlClient, HINT_TO_MSSQL_ATTR
+
+from dlt.destinations.impl.synapse import capabilities
+from dlt.destinations.impl.synapse.sql_client import SynapseSqlClient
+from dlt.destinations.impl.synapse.configuration import SynapseClientConfiguration
+
+
+HINT_TO_SYNAPSE_ATTR: Dict[TColumnHint, str] = {
+    "primary_key": "PRIMARY KEY NONCLUSTERED NOT ENFORCED",
+    "unique": "UNIQUE NOT ENFORCED",
+}
+
+
+class SynapseClient(MsSqlClient, SupportsStagingDestination):
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: SynapseClientConfiguration) -> None:
+        sql_client = SynapseSqlClient(config.normalize_dataset_name(schema), config.credentials)
+        InsertValuesJobClient.__init__(self, schema, config, sql_client)
```

Comment:
what is this line about?

Reply:
The code mirrored `MsSqlClient.__init__`. While `MsSqlClient` can call `InsertValuesJobClient.__init__` through `super()`, `SynapseClient.__init__` needs to call it explicitly (because it inherits from `MsSqlClient`, and not from `InsertValuesJobClient` like `MsSqlClient` does).

That being said, I have added a new commit to simplify `SynapseClient.__init__`. Would you agree the new code is better?